test/component_fetcher_test.exs

```diff
@@ -0,0 +1,38 @@
+defmodule MozartFetcher.ComponentFetcherTest do
+  use ExUnit.Case
+
+  alias MozartFetcher.{ComponentFetcher}
+
+  describe "fetch" do
+    test "successful response" do
+      defmodule MockClient do
+        def get(_) do
+          {:ok, %HTTPoison.Response{}}
+        end
+      end
+
+      resp = ComponentFetcher.fetch("http://localhost:8082/foo/bar", MockClient)
+
+      assert resp ==
+               {:ok,
+                %HTTPoison.Response{
+                  body: nil,
+                  headers: [],
+                  request: nil,
+                  request_url: nil,
+                  status_code: nil
+                }}
+    end
+  end
+
+  test "unsuccessful response" do
+    defmodule MockClient do
```

Comment:
When defining this mock for another test I get a warning about redefining. I could rename this, but maybe there is another way to silence the warnings.